543.py

```diff
@@ -0,0 +1,31 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def diameterOfBinaryTree(self, root: TreeNode) -> int:
+        def getMaxDepth(node: TreeNode) -> int:
+            if not node:
+                return 0
+
+            if not node.left and not node.right:
```

Comment:
Line 13 is not needed to make this `getMaxDepth` function correct. If both children are `None`, line 16 (as well as the subsequent recursive calls from line 16) handles it correctly.

Line 10 is needed because a `None` node will cause `node.left` and `node.right` to be incorrect.

You may argue that line 13 enables early return and saves two more calls to the next level. My counter argument is that we need to separate concerns between the two levels of recursive calls: a `None` node should take care of itself (`if not node: return 0`); its parent should not interfere at the parent level. 

Nowadays, the benefit we get from improving readability and maintainability and thus saving developers' time outweigh the benefit we get from saving two functions calls for the machine. I want to make sure whatever template you use, it is the simplest and correct template. Simpler means easier to memorize and less chance to introduce error.